src/HotChocolate/Core/src/Types.Scalars/RegularExpressionType.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Text.RegularExpressions;
+using HotChocolate.Language;
+
+namespace HotChocolate.Types.Scalars
+{
+    /// <summary>
+    /// The Regular Expression scalar type represents a String Type following 
+    /// a regular expression pattern/format, as defined
+    /// here https://en.wikipedia.org/wiki/Regular_expression.
```

Comment:
The Regular Expression scalar type represents textual data, represented as UTF‐8 character sequences following a pattern defined as a regular expression.